### PR TITLE
zh: doc for how to evict tikv pod

### PR DESCRIPTION
**Pull request by liubog2008**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-tidb-operator) that's required for repo owners to accept my contribution.

### Which TiDB Operator version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v1.2 (TiDB Operator 1.2 versions)
- [x] v1.1 (TiDB Operator 1.1 versions)
- [ ] v1.0 (TiDB Operator 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: 
- Other reference link(s): 

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: zh/maintain-a-kubernetes-node.md [hunk 1]
```diff
@@ -6,61 +6,88 @@ aliases: ['/docs-cn/tidb-in-kubernetes/dev/maintain-a-kubernetes-node/']
...
+
+    2. 通过调整 PD 的 `max-store-down-time` 配置来增大集群所允许的 TiKV Pod 下线时间，在此时间内维护完毕并恢复 Kubernetes 节点后，所有该节点上的 TiKV Pod 会自动恢复。
+
+        {{< copyable "shell-regular" >}}
+
+        ```shell
+        pd-ctl config set max-store-down-time 60m
+        ```
+
+        调整 `max-store-down-time` 到合理的值。
```
[the remark this answers is not in the document]
Comment (DanielZhangQD) [reply zh/maintain-a-kubernetes-node.md:52]: 不会吧，只有上面的命令会被复制，cc @qiancai

File: zh/maintain-a-kubernetes-node.md [hunk 2]
```diff
@@ -6,61 +6,88 @@ aliases: ['/docs-cn/tidb-in-kubernetes/dev/maintain-a-kubernetes-node/']
...
 > **注意：**
 >
-> 长期维护节点前，需要保证 Kubernetes 集群的剩余资源足够运行 TiDB 集群。
+> 维护节点前，需要保证 Kubernetes 集群的剩余资源足够运行 TiDB 集群。
 
-## 维护 PD 和 TiDB 实例所在节点
+## 维护短期内可恢复的节点
 
-PD 和 TiDB 实例的迁移较快，可以采取主动驱逐实例到其它节点上的策略进行节点维护：
+1. 使用 `kubectl cordon` 命令防止新的 Pod 调度到待维护节点上：
```
Comment (qiancai) [inline zh/maintain-a-kubernetes-node.md:24]:
```suggestion
1. 使用 `kubectl cordon` 命令标记待维护节点为不可调度，防止新的 Pod 调度到待维护节点上：
```

File: zh/maintain-a-kubernetes-node.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (qiancai) [reply zh/maintain-a-kubernetes-node.md:52]: 是的，赞同 @csuzhangxc，其实 shell 命令前面那句话稍微改改， “调整 `max-store-down-time` 到合理的值。”这句话就可以删除了。

File: zh/maintain-a-kubernetes-node.md [hunk 3]
```diff
@@ -6,61 +6,88 @@ aliases: ['/docs-cn/tidb-in-kubernetes/dev/maintain-a-kubernetes-node/']
...
-    kubectl get pod --all-namespaces -o wide | grep ${node_name}
+    kubectl get pod --all-namespaces -o wide | grep ${node_name} | grep tikv
     ```
 
-    假如存在 TiKV 实例，请参考[维护 TiKV 实例所在节点](#维护-tikv-实例所在节点)。
+    假如存在 TiKV Pod，针对每一个 Pod：
+    
+    1. 参考[迁移 TiKV Region Leader](#迁移-tikv-region-leader) 将 Region Leader 迁移到其他 Pod。
+
+    2. 通过调整 PD 的 `max-store-down-time` 配置来增大集群所允许的 TiKV Pod 下线时间，在此时间内维护完毕并恢复 Kubernetes 节点后，所有该节点上的 TiKV Pod 会自动恢复。
```
Comment (qiancai) [inline zh/maintain-a-kubernetes-node.md:44]:
```suggestion
    2. 通过调整 PD 的 `max-store-down-time` 配置来增大集群所允许的 TiKV Pod 下线时间，在此时间内维护完毕并恢复 Kubernetes 节点后，所有该节点上的 TiKV Pod 会自动恢复。
    
    以调整 `max-store-down-time` 为 ‘60m’ 为例，请使用以下命令：
```

File: zh/maintain-a-kubernetes-node.md [hunk 4]
```diff
@@ -6,61 +6,88 @@ aliases: ['/docs-cn/tidb-in-kubernetes/dev/maintain-a-kubernetes-node/']
...
 
     {{< copyable "shell-regular" >}}
 
     ```shell
-    kubectl get pod --all-namespaces -o wide | grep ${node_name}
+    kubectl get pod --all-namespaces -o wide | grep ${node_name} | grep tikv
     ```
 
-    假如存在 TiKV 实例，请参考[维护 TiKV 实例所在节点](#维护-tikv-实例所在节点)。
+    假如存在 TiKV Pod，针对每一个 Pod：
```
Comment (qiancai) [inline zh/maintain-a-kubernetes-node.md:40]:
```suggestion
    假如存在 TiKV Pod，针对每一个 Pod，进行以下操作：
```

File: zh/maintain-a-kubernetes-node.md [hunk 5]
```diff
@@ -125,58 +183,66 @@ pd-ctl -d config set max-store-down-time 10m
...
 
     {{< copyable "shell-regular" >}}
 
     ```shell
-    tkctl get -A tikv | grep ${node_name}
+    kubectl get pod --all-namespaces -o wide | grep ${node_name} | grep pd
     ```
 
-3. 使用 `pd-ctl` 主动下线 TiKV 实例。
+### 如果节点存储可迁移
```
Comment (qiancai) [inline zh/maintain-a-kubernetes-node.md:194]: 请问 “如果节点存储可迁移” 和  “如果节点存储不可迁移” 都是重调度 TiKV Pod 的步骤吗？

File: zh/maintain-a-kubernetes-node.md [hunk 6]
```diff
@@ -70,52 +97,83 @@ PD 和 TiDB 实例的迁移较快，可以采取主动驱逐实例到其它节
...
+    kubectl delete node ${node_name}
+    ```
 
-### 维护短期内不可恢复的节点
+## 重调度 PD Pod
 
-针对短期内不可恢复的节点维护，如节点长期下线等情形，需要使用 `pd-ctl` 主动通知 TiDB 集群下线对应的 TiKV 实例，再手动解除 TiKV 实例与该节点的绑定。
+针对节点长期下线等情形，需要把该节点上的 PD Pod 重调度到其他节点。
 
 1. 使用 `kubectl cordon` 命令防止新的 Pod 调度到待维护节点上：
```
Comment (qiancai) [inline zh/maintain-a-kubernetes-node.md:178]:
```suggestion
1. 使用 `kubectl cordon` 命令标记待维护节点为不可调度，防止新的 Pod 调度到待维护节点上：
```

File: zh/maintain-a-kubernetes-node.md [hunk 7]
```diff
@@ -187,46 +253,192 @@ pd-ctl -d config set max-store-down-time 10m
...
     watch kubectl -n ${namespace} get pod -o wide
     ```
 
-    假如待维护节点上还有其它 TiKV 实例，则重复同样的操作步骤直到所有的 TiKV 实例都迁移到其它节点上。
+## 重调度 TiKV Pod
 
-8. 确认节点不再有 TiKV 实例后，再逐出节点上的其它实例：
+针对节点长期下线等情形，需要把该节点上的 TiKV Pod 重调度到其他节点。
+
+1. 使用 `kubectl cordon` 命令防止新的 Pod 调度到待维护节点上：
```
Comment (qiancai) [inline zh/maintain-a-kubernetes-node.md:276]:
```suggestion
1. 使用 `kubectl cordon` 命令标记待维护节点为不可调度，防止新的 Pod 调度到待维护节点上：
```

File: zh/maintain-a-kubernetes-node.md [hunk 8]
```diff
@@ -187,46 +253,192 @@ pd-ctl -d config set max-store-down-time 10m
...
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    pd-ctl scheduler remove evict-leader-scheduler-${ID}
+    ```
+
+## 迁移 PD Leader
+
+1. 查看 PD Leader
```
Comment (qiancai) [inline zh/maintain-a-kubernetes-node.md:402]:
```suggestion
1. 查看 PD Leader：
```

File: zh/maintain-a-kubernetes-node.md (hunk 5 above)
Comment (liubog2008) [reply zh/maintain-a-kubernetes-node.md:194]: 是的，两种情况

Comment (qiancai) [reply zh/maintain-a-kubernetes-node.md:196]: 是否可以将“使用 `kubectl cordon` 命令标记待维护节点为不可调度” 和 “查看待维护节点上的 PD Pod“ 这两个步骤挪到下面的这两个章节里呢，这样步骤呈现更清晰？

如果节点存储可自动迁移

如果节点存储不可自动迁移

Comment (liubog2008) [reply zh/maintain-a-kubernetes-node.md:196]: 重复一遍的意思么？

Comment (qiancai) [reply zh/maintain-a-kubernetes-node.md:196]: 是的，重复写两个步骤，这样这两种情况各自的内容看起来会更集中些

File: zh/maintain-a-kubernetes-node.md [hunk 9]
```diff
@@ -187,46 +272,210 @@ pd-ctl -d config set max-store-down-time 10m
...
     watch kubectl -n ${namespace} get pod -o wide
     ```
 
-    假如待维护节点上还有其它 TiKV 实例，则重复同样的操作步骤直到所有的 TiKV 实例都迁移到其它节点上。
+## 重调度 TiKV Pod
 
-8. 确认节点不再有 TiKV 实例后，再逐出节点上的其它实例：
+### 如果节点存储可迁移
+
+针对节点长期下线等情形，需要把该节点上的 TiKV Pod 重调度到其他节点。
```
Comment (qiancai) [inline zh/maintain-a-kubernetes-node.md:295]:
```suggestion
针对节点长期下线等情形，需要将节点上的 TiKV Pod 重调度到其他节点。

### 如果节点存储可自动迁移

如果节点存储可以自动迁移（比如使用 EBS），你不需要删除整个 TiKV Store，只需要迁移 Region Leader 到其他 Pod 后删除原来的 Pod 就可以实现重调度。
```

File: zh/maintain-a-kubernetes-node.md [hunk 10]
```diff
@@ -187,46 +272,210 @@ pd-ctl -d config set max-store-down-time 10m
...
     {{< copyable "shell-regular" >}}
 
     ```shell
-    kubectl get pod --all-namespaces | grep ${node_name}
+    kubectl get pod --all-namespaces -o wide | grep ${node_name} | grep tikv
     ```
 
-10. 最后（可选），假如是长期下线节点，建议将节点从 Kubernetes 集群中删除：
+如果节点存储可以自动迁移，比如使用 EBS，则不需要删除整个 TiKV Store，只需要迁移 Region Leader 并删除 Pod。
+
```
Comment (qiancai) [inline zh/maintain-a-kubernetes-node.md:314]:
```suggestion
```

File: zh/maintain-a-kubernetes-node.md [hunk 11]
```diff
@@ -187,46 +272,210 @@ pd-ctl -d config set max-store-down-time 10m
...
+2. 查看待维护节点上的 TiKV Pod：
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    kubectl get pod --all-namespaces -o wide | grep ${node_name} | grep tikv
+    ```
+
+如果节点存储不可以自动迁移，比如使用本地存储，则需要删除整个 TiKV Store。
+
```
Comment (qiancai) [inline zh/maintain-a-kubernetes-node.md:362]:
```suggestion
```

File: zh/maintain-a-kubernetes-node.md [hunk 12]
```diff
@@ -187,46 +272,210 @@ pd-ctl -d config set max-store-down-time 10m
...
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    pd-ctl scheduler remove evict-leader-scheduler-${ID}
+    ```
+
+### 如果节点存储不可迁移
+
+针对节点长期下线等情形，需要把该节点上的 TiKV Pod 重调度到其他节点。
```
Comment (qiancai) [inline zh/maintain-a-kubernetes-node.md:343]:
```suggestion
### 如果节点存储不可自动迁移

如果节点存储不可以自动迁移（比如使用本地存储），你需要删除整个 TiKV Store 以实现重调度。
```

File: zh/maintain-a-kubernetes-node.md [hunk 13]
```diff
@@ -70,113 +99,169 @@ PD 和 TiDB 实例的迁移较快，可以采取主动驱逐实例到其它节
...
 
     {{< copyable "shell-regular" >}}
 
     ```shell
-    tkctl get -A tikv | grep ${node_name}
+    kubectl get pod --all-namespaces -o wide | grep ${node_name} | grep pd
     ```
 
-3. 使用 `pd-ctl` 主动下线 TiKV 实例。
+如果节点存储可以自动迁移，比如使用 EBS，则不需要删除 PD Member，只需要迁移 Leader 并删除 Pod。
```
Comment (qiancai) [inline zh/maintain-a-kubernetes-node.md:199]:
```suggestion
```

File: zh/maintain-a-kubernetes-node.md [hunk 14]
```diff
@@ -70,113 +99,169 @@ PD 和 TiDB 实例的迁移较快，可以采取主动驱逐实例到其它节
...
+
+    ```shell
+    kubectl delete node ${node_name}
+    ```
+
+## 重调度 PD Pod
+
+### 如果节点存储可迁移
+
+针对节点长期下线等情形，需要把该节点上的 PD Pod 重调度到其他节点。
```
Comment (qiancai) [inline zh/maintain-a-kubernetes-node.md:180]:
```suggestion
针对节点长期下线等情形，需要将该节点上的 PD Pod 重调度到其他节点。

### 如果节点存储可自动迁移

如果节点存储可以自动迁移（比如使用 EBS），你不需要删除 PD Member，只需要迁移 Leader 到其他 Pod 后删除原来的 Pod 就可以实现重调度。


```

File: zh/maintain-a-kubernetes-node.md [hunk 15]
```diff
@@ -70,113 +99,169 @@ PD 和 TiDB 实例的迁移较快，可以采取主动驱逐实例到其它节
...
 
+2. 查看待维护节点上的 PD Pod：
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    kubectl get pod --all-namespaces -o wide | grep ${node_name} | grep pd
+
+如果节点存储不可以自动迁移，比如使用本地存储，则需要删除 PD Member。
+
```
Comment (qiancai) [inline zh/maintain-a-kubernetes-node.md:238]:
```suggestion
```

File: zh/maintain-a-kubernetes-node.md [hunk 16]
```diff
@@ -70,113 +99,169 @@ PD 和 TiDB 实例的迁移较快，可以采取主动驱逐实例到其它节
...
 
     ```shell
-    kubectl get tc ${CLUSTER_NAME} -ojson | jq '.status.tikv.stores | .[] | select ( .podName == "${POD_NAME}" ) | .id'
+    watch kubectl -n ${namespace} get pod -o wide
     ```
 
-    下线实例：
+### 如果节点存储不可迁移
+
+针对节点长期下线等情形，需要把该节点上的 PD Pod 重调度到其他节点。
```
Comment (qiancai) [inline zh/maintain-a-kubernetes-node.md:220]:
```suggestion
### 如果节点存储不可自动迁移

如果节点存储不可以自动迁移（比如使用本地存储），你需要删除 PD Member 以实现重调度。
```

File: zh/maintain-a-kubernetes-node.md [hunk 17]
```diff
@@ -70,113 +99,169 @@ PD 和 TiDB 实例的迁移较快，可以采取主动驱逐实例到其它节
...
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    kubectl delete node ${node_name}
+    ```
+
+## 重调度 PD Pod
+
+针对节点长期下线等情形，需要将该节点上的 PD Pod 重调度到其他节点。
```
Comment (liubog2008) [inline zh/maintain-a-kubernetes-node.md:178]:
```suggestion
针对节点长期下线等情形，为了尽可能减少业务受到的影响，可以将该节点上的 PD Pod 预先调度到其他节点。
```

File: zh/maintain-a-kubernetes-node.md [hunk 18]
```diff
@@ -187,46 +272,208 @@ pd-ctl -d config set max-store-down-time 10m
...
     {{< copyable "shell-regular" >}}
 
     ```shell
     watch kubectl -n ${namespace} get pod -o wide
     ```
 
-    假如待维护节点上还有其它 TiKV 实例，则重复同样的操作步骤直到所有的 TiKV 实例都迁移到其它节点上。
+## 重调度 TiKV Pod
+
+针对节点长期下线等情形，需要将节点上的 TiKV Pod 重调度到其他节点。
```
Comment (liubog2008) [inline zh/maintain-a-kubernetes-node.md:293]:
```suggestion
针对节点长期下线等情形，为了尽可能减少业务受到的影响，可以将该节点上的 TiKV Pod 预先调度到其他节点。
```

File: zh/maintain-a-kubernetes-node.md [hunk 19]
```diff
@@ -6,61 +6,90 @@ aliases: ['/docs-cn/tidb-in-kubernetes/dev/maintain-a-kubernetes-node/']
...
 
-    假如存在 TiKV 实例，请参考[维护 TiKV 实例所在节点](#维护-tikv-实例所在节点)。
+    假如存在 TiKV Pod，针对每一个 Pod，进行以下操作：
+    
+    1. 参考[迁移 TiKV Region Leader](#迁移-tikv-region-leader) 将 Region Leader 迁移到其他 Pod。
 
-2. 使用 `kubectl cordon` 命令防止新的 Pod 调度到待维护节点上：
+    2. 通过调整 PD 的 `max-store-down-time` 配置来增大集群所允许的 TiKV Pod 下线时间，在此时间内维护完毕并恢复 Kubernetes 节点后，所有该节点上的 TiKV Pod 会自动恢复。
+    
+    以调整 `max-store-down-time` 为 ‘60m’ 为例，请使用以下命令：
```
Comment (qiancai) [inline zh/maintain-a-kubernetes-node.md:46]:
```suggestion
        以调整 `max-store-down-time` 为 `60m` 为例，请使用以下命令：
```

File: zh/maintain-a-kubernetes-node.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (qiancai) [reply zh/maintain-a-kubernetes-node.md:54]:
```suggestion
```

File: zh/maintain-a-kubernetes-node.md [hunk 20]
```diff
@@ -6,61 +6,90 @@ aliases: ['/docs-cn/tidb-in-kubernetes/dev/maintain-a-kubernetes-node/']
...
+
+        {{< copyable "shell-regular" >}}
+
+        ```shell
+        pd-ctl config set max-store-down-time 60m
+        ```
+
```
Comment (qiancai) [inline zh/maintain-a-kubernetes-node.md:53]:
```suggestion
```

**Review by qiancai — APPROVED** [review]

**Pull request merged**